### PR TITLE
feat: upgrade deprecated node version

### PR DESCRIPTION
**Pull request by MichaelWalker-git**
### Description
- Upgrades deprecated Node version 12 to supported version 18

Fixes # 
---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: typescript/lambda-manage-s3-event-notification/lib/shared-resources-stack.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export class SharedStack extends cdk.Stack {
     this.bucketName = bucket.bucketName;
 
     const fn = new lambda.Function(this, 'S3EventNotificationsLambda', {
-      runtime: lambda.Runtime.NODEJS_12_X,
+      runtime: lambda.Runtime.NODEJS_18_X,
```
[the remark this answers is not in the document]
Comment (MichaelWalker-git) [reply typescript/lambda-manage-s3-event-notification/lib/shared-resources-stack.ts:22]: I apologise for that oversight. I shall update the lambda to v3 and test it locally. I was just passing by and thought it was a quick win.

Ill have an update PR soon

**Review by kaizio — DISMISSED** [review]
Temp check

**Comment by kaizio** [comment]
@Mergifyio update

File: typescript/lambda-manage-s3-event-notification/lib/shared-resources-stack.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (MichaelWalker-git) [reply typescript/lambda-manage-s3-event-notification/lib/shared-resources-stack.ts:22]: I have updated the lambda with the SDK v3 references.

**Comment by MichaelWalker-git** [comment]
Updated the lambda with V3 SDK

**Review by kaizio — APPROVED** [review]

**Pull request merged**